service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/EhrExtractStatusService.java

```diff
@@ -129,4 +126,62 @@ public void updateEhrExtractStatusCore(SendEhrExtractCoreTaskDefinition sendEhrE
             throw new EhrExtractException("EHR Extract Status was not updated with Extract Core Message.");
         }
     }
+
+    public void updateEhrExtractStatusAccessDocumentPatientId(GetGpcDocumentReferencesTaskDefinition patientIdentifierTaskDefinition,
+            String patientId) {
+        Query query = createQueryForConversationId(patientIdentifierTaskDefinition.getConversationId());
+
+        Update update = createUpdateWithUpdatedAt();
+        if (!patientId.isBlank()) {
+            update.set(DOCUMENT_PATIENT_ID, patientId);
+        } else {
+            update.set(DOCUMENT_PATIENT_ID, null);
+        }
+        update.set(GPC_DOCUMENTS, new ArrayList<>());
+        UpdateResult updateResult = mongoTemplate.updateFirst(query, update, EhrExtractStatus.class);
+
+        if (updateResult.getModifiedCount() != 1) {
+            throw new EhrExtractException("EHR Extract Status was not updated with patientId");
+        }
+    }
+
+    public void updateEhrExtractStatusAccessDocumentDocumentReferences(
+            GetGpcDocumentReferencesTaskDefinition documentReferencesTaskDefinition,
+            List<String> urls) {
+        Query query = createQueryForConversationId(documentReferencesTaskDefinition.getConversationId());
+
+        Update update = createUpdateWithUpdatedAt();
+        Instant now = Instant.now();
+        urls.forEach(url -> {
+            update.addToSet(GPC_DOCUMENTS, new EhrExtractStatus.GpcAccessDocument.GpcDocument(
```

Comment:
Prefer builders over constructors with multiple args to avoid bugs caused by misplaced positional arguments

`...GpcDocument.builder()...build()`